apps/emqx/src/emqx_frame.erl

```diff
@@ -100,14 +100,13 @@ parse(<<Type:4, Dup:1, QoS:2, Retain:1, Rest/binary>>,
     StrictMode andalso validate_header(Type, Dup, QoS, Retain),
     Header = #mqtt_packet_header{type   = Type,
                                  dup    = bool(Dup),
-                                 qos    = QoS,
+                                 qos    = case fixqos(Type, QoS) of
+                                              QoS -> QoS;
+                                              FixedQoS -> FixedQoS
+                                          end,
```

Comment:
clause 104 and 105 are not necessary. just 
```suggestion
                                 qos    = fixqos(Type, QoS),
```

Reply:
Already changed. Please review again.